projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-site-settings-endpoint.php

```diff
@@ -365,6 +367,8 @@ public function get_settings_response() {
 						'jetpack_sync_non_public_post_stati' => (bool) Jetpack_Options::get_option( 'sync_non_public_post_stati' ),
 						'jetpack_relatedposts_allowed'     => (bool) $this->jetpack_relatedposts_supported(),
 						'jetpack_relatedposts_enabled'     => (bool) $jetpack_relatedposts_options['enabled'],
+						'jetpack_relatedposts_show_context' => isset( $jetpack_relatedposts_options['show_context'] ) && $jetpack_relatedposts_options['show_context'],
+						'jetpack_relatedposts_show_date'   => isset( $jetpack_relatedposts_options['show_date'] ) && $jetpack_relatedposts_options['show_date'],
```

Comment:
Should we cast those 2 as booleans like we do above and below, for consistency?

Reply:
Sure. I suppose it is there only for human readability, right? I actually had the casting and removed it because I thought there was no need for it. 😅

Reply:
Btw, I tested the PR with and also without casting - as I started testing it when the `(bool)`s were still there. 🙂

Reply:
BTW the casting done on the lines below seems incorrect to me too. It only casts the result of `isset( $jetpack_relatedposts_options['show_headline'] )`. The actual value that is being returned ends up being "1", "0" or false.

Reply:
> BTW the casting done on the lines below seems incorrect to me too

That's a good point actually, nice catch! We could update all those to be correct, and to use your format? We wouldn't even need to cast as booleans then, as you pointed out. I was mostly concerned about consistency, with your original approach the (bool) isn't really necessary.

Could you update all those in this PR, or in a follow-up PR?

Reply:
For that matter, `isset( $foo ) && (bool) $foo` is the same as `! empty( $foo )` if you want to simplify the code a bit.

Reply:
@jeherve let me do it with this one. 🙂 

@anomiex good point. I will do as per your suggestion. 👍